addons/mail/static/src/models/messaging_menu/messaging_menu.js

```diff
@@ -31,40 +31,31 @@ function factory(dependencies) {
          */
         toggleOpen() {
             this.update({ isOpen: !this.isOpen });
-            this.env.messaging.refreshIsNotificationPermissionDefault();
+            this.messaging.refreshIsNotificationPermissionDefault();
+            if (this.isOpen) {
+                // populate some needaction messages on threads.
+                this.messaging.inbox.cache.update({ isCacheRefreshRequested: true });
+            }
         }
 
         //----------------------------------------------------------------------
         // Private
         //----------------------------------------------------------------------
 
-        /**
-         * @private
-         */
-        _computeInboxMessagesAutoloader() {
-            if (!this.isOpen) {
-                return;
-            }
-            const inbox = this.env.messaging.inbox;
-            if (!inbox || !inbox.cache) {
-                return;
-            }
-            // populate some needaction messages on threads.
-            inbox.cache.update({ isCacheRefreshRequested: true });
-        }
-
         /**
          * @private
          * @returns {integer}
          */
         _computeCounter() {
-            if (!this.env.messaging) {
+            if (!this.messaging) {
                 return 0;
             }
-            const inboxCounter = this.env.messaging.inbox ? this.env.messaging.inbox.counter : 0;
-            const unreadChannelsCounter = this.env.messaging.allPinnedChannels.filter(
-                channel => channel.localMessageUnreadCounter > 0
-            ).length;
+            const inboxCounter = this.messaging.inbox ? this.messaging.inbox.counter : 0;
+            const unreadChannelsCounter = this.env.models['mail.thread'].all(channel => (
+                channel.model === 'mail.channel',
+                channel.isPinned,
+                channel.localMessageUnreadCounter > 0
+            )).length;
```

Comment:
@seb-odoo 
Looks like a mistake, each expressions in `all()` inner-function should be combined with logical operator `&` and not `,` (comma).

Reply:
Good catch, it doesn't really matter functionally because only the last condition actually matters and it is luckily what is returned by a comma statement.

(`localMessageUnreadCounter` is computed to 0 in non channels, and having knowledge of messages + a non 0 counter in non-pinned channels is very rare and not covered by any test)

I'll still fix it tomorrow to be correct of course.

Reply:
https://github.com/odoo/odoo/pull/74742